dist/route-recognizer.js

```diff
@@ -153,7 +153,7 @@
     function $$route$recognizer$$DynamicSegment(name) { this.name = name; }
     $$route$recognizer$$DynamicSegment.prototype = {
       eachChar: function(currentState) {
-        return currentState.put({ invalidChars: "/", repeat: true });
+        return currentState.put({ validChars: undefined, invalidChars: "/", repeat: true });
```

Comment:
We should be pedantic about the key ordering, although the transitions have some simple reductions that can reduce several different transition orderings into 1 stable order. Lets just always order keys alphabetically, that way we can be sure.